recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,237 @@
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mongoc.org/"
+    description = "A Cross Platform MongoDB Client Library for C"
+    topics = ("conan", "libbson", "libmongoc", "mongo", "mongodb", "database", "db")
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package", "pkg_config"
+    short_paths = True
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ssl": [False, "darwin", "windows", "openssl", "libressl"],
+        "with_sasl": [False, "sspi", "cyrus"],
+        "with_snappy": [True, False],
+        "with_zlib": [True, False],
+        "with_zstd": [True, False],
+        "with_icu": [True, False],
+        "srv": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": "openssl",
+        "with_sasl": False,
+        "with_snappy": True,
+        "with_zlib": True,
+        "with_zstd": True,
+        "with_icu": True,
+        "srv": True
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.with_sasl = "sspi"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+        if self.options.with_ssl == "darwin" and not tools.is_apple_os(self.settings.os):
+            raise ConanInvalidConfiguration("with_ssl=darwin only allowed on Apple os family")
+        if self.options.with_ssl == "windows" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("with_ssl=windows only allowed on Windows")
+        if self.options.with_sasl == "sspi" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("with_sasl=sspi only allowed on Windows")
+
+    def requirements(self):
+        if self.options.with_ssl == "openssl":
+            self.requires("openssl/1.1.1h")
+        elif self.options.with_ssl == "libressl":
+            self.requires("libressl/3.2.0")
+        if self.options.with_sasl == "cyrus":
+            self.requires("cyrus-sasl/2.1.27")
+        if self.options.with_snappy:
+            self.requires("snappy/1.1.8")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5")
+        if self.options.with_icu:
+            self.requires("icu/68.1")
+
+    def build_requirements(self):
+        if self.options.with_ssl == "libressl" or self.options.with_zstd:
+            self.build_requires("pkgconf/1.7.3")
```

Comment:
This requirement works on windows? I thought we needs msys for windows 

🤓 For my own learning

Reply:
I can disable it for now and enable after #2933 merged

Reply:
pkgconf doesn't need msys2 on windows, so it's fine like this.